state/application.go

```diff
@@ -1337,6 +1337,39 @@ func (a *Application) GetScale() int {
 	return a.doc.DesiredScale
 }
 
+// ChangeScale alters the existing scale by the provided change amount, returning the new amount.
+// This is used on CAAS models.
+func (a *Application) ChangeScale(scaleChange int) (int, error) {
+	newScale := a.doc.DesiredScale + scaleChange
+	if newScale < 1 {
+		return a.doc.DesiredScale, errors.NotValidf("cannot remove more units than currently exist")
+	}
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt > 0 {
```

Comment:
Need to redo the check that resulting scale is >=0 since if attempt > 0, the scale may have changed underneath us